### ISSUE TITLE
Broken cross reference links in manual in version 1.34

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/qpa#81]
E.g. `qpa/doc/chap6.html` contains `<a href="/Users/oyvinso/gap-4.11.0/doc/ref/chap62.html#X818DE6C57D1A4B33">`

Note that [ReleaseTools](https://github.com/gap-system/ReleaseTools) works hard to avoid this kind of problem -- one trick is that it uses `PushOptions(rec(relativePath:="../../.."));` before doing `Read("makedoc.g")` when building the documentation for a release.  But that in turn is an AutoDoc feature, which you don't seem to be using to generate the documentation (it is useful for that even if the manual is pure GAPDoc). Anyway, I think all this ends up doing is pass "../../.." as final argument to `MakeGAPDocDoc` ?

(ReleaseTools also scans for such issues and will refuse to make a release if the manual contains such absolute links.)

**Comment by fingolfin** [comment]
This is also broken on your website in the online documentation. Go to https://oyvinso.folk.ntnu.no/QPA/qpa/doc/chap6.html#X8710CC447F1F7B17 and try one of the two "References: ..." links (though admittedly it'd be broken either way if not placed next to a GAP manual.

But it is also broken in our official copy of the manual on the GAP website, see https://docs.gap-system.org/pkg/qpa/doc/chap6_mj.html#X8710CC447F1F7B17

It would be great if this was fixed in the next release. I'd be happy to be available as a "consultant" for that. (We are also looking into creating a new "release workflow" for use with GitHub Actions that would take care of this for you... we'll see if we manage)

**Comment by sunnyquiver** [comment]
I am trying to use ReleaseTools to make a release. In the final stage running 

```
qpa % /ReleaseTools/release-gap-package
Using PYTHON = '/opt/anaconda3/bin/python'
Using GAP = '/usr/local/bin/gap'
Extracting information from PackageInfo.g... /ReleaseTools/release-gap-package: line 307: gap:: command not found
```
Then I get the above error message, which I have no idea means. Any suggestions?

**Comment by fingolfin** [comment]
OK, odd, I'll investigate.

But in the meantime, just to double check to avoid me chasing down the wrong rabbit hole:
1. presumably `/usr/local/bin/gap` is the correct GAP executable?
2. Is this the latest `release-gap-package` script (it was last updated a year ago, 2024-03-26, and has SHA256 checksum `e4f41d212a91d4f90f27626d183bee151c0c2a9f0094fea6726fad9f605fcdd5`
3. you added `gh-pages` to the repository, this seems to be a "git submodule" markers, but there is no `.gitmodules` file -- so was this perhaps an accident? if so, I suggest removing it, and adding both `gh-pages/` and `tmp/` to your `.gitignore` file

**Comment by sunnyquiver** [comment]
1. Yes, /usr/local/bin/gap is where the correct GAP executable is. However, in .zshrc the following is set
      alias gap="/usr/local/bin/gap -l /Users/username/gap-4.14.0/"
2. Yes, the latest available is downloaded from GitHub.
3. I have tried to follow the instructions, so I have not tried to do anything out of the ordinary. 

The gap-error.log in the tmp directory says:
         gap: hmm, I cannot find 'lib/init.g' maybe use option '-l <gaproot>'?
Is this helpful?

**Comment by fingolfin** [comment]
Try setting `GAP="/usr/local/bin/gap -l /Users/username/gap-4.14.0/` before running the release scripts

**Comment by sunnyquiver** [comment]
Commands executed:

```
qpa% alias GAP="/usr/local/bin/gap -l /Users/username/gap-4.14.0/"
qpa% ReleaseTools/release-gap-package
```

Output:
```
Using PYTHON = '/Users/username/opt/anaconda3/bin/python'
Using GAP = '/usr/local/bin/gap'
Extracting information from PackageInfo.g...
ReleaseTools/release-gap-package: line 307: gap:: command not found
```

The gap-error.log is saying:

`gap: hmm, I cannot find 'lib/init.g' maybe use option '-l \<gaproot\>'?`

**Comment by fingolfin** [comment]
No, that's not what I meant: programs can't "see" aliases you defined in your shell. As a fallback, you can set an environment variable `GAP` (and similar `PYTHON`) to contain the full command needed to execute GAP. Then you can do this:
```
GAP="/usr/local/bin/gap -l /Users/username/gap-4.14.0/" release-gap-package
```
or equivalently (at least in bash):
```
export GAP="/usr/local/bin/gap -l /Users/username/gap-4.14.0/"
release-gap-package
```

A cleaner solution that will also help all kinds of other scripts that want to invoke is to just make sure executing `/usr/local/bin/gap` works on its own... I don't know what it contains, but either of these should work:
1. have it be a symlink to `/Users/username/gap-4.14.0/gap`
2. have it be a little shell script that essentially invokes `/Users/username/gap-4.14.0/gap` but passes on arguments, e.g. like this:
```
#!/bin/sh
exec "/Users/username/gap-4.14.0//gap" "$@"
```

**Comment by sunnyquiver** [comment]
Command given and output:
```
qpa % GAP="/usr/local/bin/gap -l /Users/username/gap-4.14.0/" ReleaseTools/release-gap-package

Using PYTHON = '/Users/username/opt/anaconda3/bin/python'
ERROR: could not find GAP (perhaps set the GAP environment variable?)
```
No updated of the the tmp/gap-error.log.

**Comment by fingolfin** [comment]
Good so we are getting somewhere: it detected the unresolved references that this issue is complaining about :-). Now we just need to help it resolve these. I know two (soon three...) ways to resolve this:

1. ensure your package git repository is inside `GAPROOT_DIR/pkg/qpa` (I assume it isn't for you)
2. use a hack to ensure GAPDoc can find the GAP documentation at `../..`:
   - `cd` into the clone of `qpa` you want to release from
   - create a symlink to the GAP `doc` dir in the right place: `ln -s /Users/username/gap-4.14.0/doc ../..`
   - try again
3. (... and hopefully we'll get that GitHub action for doing releases finally working tomorrow which will take care of this and other stuff)

**Comment by sunnyquiver** [comment]
I have removed the absolute references to the GAP-manual to see if script would execute, but there are still problems. 

```
qpa % /Users/username/gap-4.14.0/pkg/ReleaseTools/release-gap-package 
Using PYTHON = '/Users/username/opt/anaconda3/bin/python'
Using GAP = ''
Extracting information from PackageInfo.g...
/Users/username/gap-4.14.0/pkg/ReleaseTools/release-gap-package: line 307: /Users/username/gap-4.14.0/pkg/ReleaseTools/release-gap-package:: No such file or directory
```
Any progress here, or this an abandoned project?

**Comment by fingolfin** [comment]
Which project? ReleaseTools? It works fine for a dozen users *shrug*

 The problem is ultimately in your setup. ReleaseTools refuses to release a tarball with a broken manual in it, which is a feature, not a bug.

Still, I'd be happy to try and help you sort this out. Alas, debugging this via relay and issue comments is not very fruitful unfortunately. Most likely a 15 minute Zoom call would work better.

Ultimately a solution using GitHub actions would be best because it is much easier to get such things sorted once in a fixed environment. Unfortunately we die not manage to complete it during GAP days. @stertooy did some really nice work after GAP days but for now it all has stalled, and I am busy with other stuff, I am afraid
